feature-libs/product/configurators/rulebased/components/attribute/product-card/configurator-attribute-product-card.component.spec.ts

```diff
@@ -0,0 +1,216 @@
+      [createImage('url', 'alt')],
+      1,
+      false,
+      '1111-2222',
+      'Lorem Ipsum Dolor'
+    );
+
+    spyOn(component, 'onHandleSelect').and.callThrough();
+    spyOn(component, 'onHandleDeselect').and.callThrough();
+
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should button be enabled when disabledAction = false & selected = false', () => {
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+    expect(button.disabled).toBe(false);
+  });
+
+  it('should button be enabled when disabledAction = false & selected = true', () => {
+    component.product.selected = true;
+
+    fixture.detectChanges();
+
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+    expect(button.disabled).toBe(false);
+  });
+
+  it('should button be disabled when disabledAction = true & selected = true', () => {
+    component.disabledAction = true;
+    component.product.selected = true;
+
+    fixture.detectChanges();
+
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+
+    expect(button.disabled).toBe(true);
+  });
+
+  it('should button be called with onHandleSelect', () => {
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+    button.click();
+
+    fixture.detectChanges();
+
+    expect(component.onHandleSelect).toHaveBeenCalled();
+  });
+
+  it('should button be called with onHandleDeselect', () => {
+    component.product.selected = true;
+
+    fixture.detectChanges();
+
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+
+    button.click();
+
+    fixture.detectChanges();
+
+    expect(component.onHandleDeselect).toHaveBeenCalled();
+  });
+
+  it('should button be select when multiSelect = false & selected = false', () => {
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+
+    expect(button.innerText).toContain('configurator.button.select');
+  });
+
+  it('should button be deselect when multiSelect = false & selected = true', () => {
+    component.product.selected = true;
+
+    fixture.detectChanges();
+
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+
+    expect(button.innerText).toContain('configurator.button.deselect');
+  });
+
+  it('should button be add when multiSelect = true & selected = false', () => {
+    component.multiSelect = true;
+    component.product.selected = false;
+
+    fixture.detectChanges();
+
+    const button = fixture.debugElement.query(By.css('button.btn'))
+      .nativeElement;
+
+    expect(button.innerText).toContain('configurator.button.add');
+  });
+
+  it('should button be remove when multiSelect = true & selected = true', () => {
```

Comment:
same as above

Reply:
also, `button be remove` doesn't make sense

Reply:
renamed